### ISSUE TITLE
NuGet.exe 5.9.1 throws "Object reference not set to an instance of an object." exception

### ISSUE THREAD

**Issue by gc-deniszmitrichenko** [NuGet/Home#10818]
## Details about Problem

NuGet product used: NuGet.exe

Product version: 5.9.1

Worked before? If so, with which NuGet version: 5.7.1

## Repro steps and/or sample project

1. Use attached txt file's content for nuspec file (you can just rename it to System.Windows.Extensions.nuspec)

[System.Windows.Extensions.txt](https://github.com/NuGet/Home/files/6406312/System.Windows.Extensions.txt)

2. Execute such command in command prompt:
`nuget.exe pack System.Windows.Extensions.nuspec -version "1.0.0" -OutputDirectory "C:\OutputDir"`

You can use any dll (instead of System.Windows.Extensions.dll) with valid path in the nuspec file and any valid name for the nuspec file itself.
I upgraded nuget.exe from 5.7.1 to 5.9.1 on my build server and it stopped to work:

![image](https://user-images.githubusercontent.com/61823719/116714543-8db1b680-a9de-11eb-8b4e-bcab45c2fe5e.png)

### Verbose Logs

NuGet Version: 5.9.1.11
Attempting to build package from 'System.Windows.Extensions.nuspec'.
Object reference not set to an instance of an object.
System.NullReferenceException: Object reference not set to an instance of an object.
   at NuGet.Packaging.PackageBuilder.<>c.<ValidateReferenceAssemblies>b__173_1(NuGet.Frameworks1204671.NuGetFramework groupFramework)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.HashSet`1.UnionWith(IEnumerable`1 other)
   at System.Collections.Generic.HashSet`1..ctor(IEnumerable`1 collection, IEqualityComparer`1 comparer)
   at NuGet.Packaging.PackageBuilder.ValidateReferenceAssemblies(IEnumerable`1 files, IEnumerable`1 packageAssemblyReferences)
   at NuGet.Packaging.PackageBuilder.Save(Stream stream)
   at NuGet.Commands.PackCommandRunner.BuildPackage(PackageBuilder builder, String outputPath, Boolean symbolsPackage)
   at NuGet.Commands.PackCommandRunner.BuildFromNuspec(String path)
   at NuGet.CommandLine.PackCommand.ExecuteCommand()
   at NuGet.CommandLine.Command.ExecuteCommandAsync()
   at NuGet.CommandLine.Command.Execute()
   at NuGet.CommandLine.Program.MainCore(String workingDirectory, String[] args)

**Comment by kartheekp-ms** [comment]
@gc-deniszmitrichenko - I think you are hitting the [known issue in 5.9 release](https://github.com/NuGet/docs.microsoft.com-nuget/blob/main/docs/release-notes/NuGet-5.9.md#nuget-59-pack-raises-null-reference-exception---10685). We have just merged a docs PR but it is not yet published to Production. Hopefully, the workaround suggested in the doc will unblock you. Given above details, I am closing this issue now. Please feel free to re-open it if you are unblocked.

Duplicate of #10685
